### ISSUE TITLE
Checkout: Update headings and labels to use sentence case

### ISSUE THREAD

**Issue by jancavan** [Automattic/wp-calypso#41768]
For consistency around our product, we are updating our card headings and form labels to use sentence case in this PR: https://github.com/Automattic/wp-calypso/pull/41647

Previously, it was a mix of sentence and title case with no guidelines as to what to use.

We would like these to be updated in the checkout pages as well.

In this example, "Domain Registration" should be "Domain registration":
![Screen Shot 2020-05-01 at 11 44 54 AM](https://user-images.githubusercontent.com/4924246/81016598-9b594e00-8e15-11ea-8b9f-97dd4e5b411d.png)

**Other examples include:**

- .blog Domain Registration
- Domain Registration
- .live Domain Registration
- Support Session
- Premium Theme: Ovation

**These should be updated to:**

- .blog domain registration
- Domain registration
- .live domain registration
- WordPress.com Personal
- Support session
- Premium theme: Ovation

For more info: p2-pbAPfg-dH

TLDR;

> Page titles remain in title case.
> All other titles can be in sentence case, except when referring to an official/branded feature or service name.
> Taglines, subheadings, descriptions, etc. should always end with a period (and, ideally, be actual sentences rather than fragments).

cc @rickybanister @lcollette @gwwar @kwight
